pkg/modifycontroller/controller.go

```diff
@@ -0,0 +1,378 @@
+	// 4. PVC is in Bound state
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeAttributesClass) {
+		oldVacName := oldPVC.Spec.VolumeAttributesClassName
+		newVacName := newPVC.Spec.VolumeAttributesClassName
+		if newVacName != nil && *newVacName != "" && *newVacName != *oldVacName && oldPVC.Status.Phase == v1.ClaimBound {
+			volumeObj, exists, err := ctrl.volumes.GetByKey(oldPVC.Spec.VolumeName)
+			if err != nil {
+				klog.Errorf("Get PV %q of pvc %q failed: %v", oldPVC.Spec.VolumeName, klog.KObj(oldPVC), err)
+				return
+			}
+			if !exists {
+				klog.InfoS("PV bound to PVC not found", "PV", oldPVC.Spec.VolumeName, "PVC", klog.KObj(oldPVC))
+				return
+			}
+
+			pv, ok := volumeObj.(*v1.PersistentVolume)
+			if !ok {
+				klog.Errorf("expected volume but got %+v", volumeObj)
+				return
+			}
+
+			_, _, err, _ = ctrl.modify(newPVC, pv)
+			if err != nil {
+				klog.Errorf("failed to modify volume at pvc update %v", err)
+				return
+			}
+		} else {
+			klog.V(4).InfoS("No need to modify PVC", "PVC", klog.KObj(newPVC))
+		}
+	}
+}
+
+func (ctrl *modifyController) deletePVC(obj interface{}) {
+	objKey, err := getObjectKey(obj)
+	if err != nil {
+		return
+	}
+	ctrl.claimQueue.Forget(objKey)
+}
+
+// Patches a given PVC with changes from newPVC. If addResourceVersionCheck is true
+// then a version check is added to the patch to ensure that we are not patching
+// old(and possibly outdated) PVC objects.
+func (ctrl *modifyController) patchClaim(oldPVC, newPVC *v1.PersistentVolumeClaim, addResourceVersionCheck bool) (*v1.PersistentVolumeClaim, error) {
+	patchBytes, err := util.GetPVCPatchData(oldPVC, newPVC, addResourceVersionCheck)
+	if err != nil {
+		return oldPVC, fmt.Errorf("can't patch status of PVC %s as generate path data failed: %v", klog.KObj(oldPVC), err)
+	}
+	updatedClaim, updateErr := ctrl.kubeClient.CoreV1().PersistentVolumeClaims(oldPVC.Namespace).
+		Patch(context.TODO(), oldPVC.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{}, "status")
+	if updateErr != nil {
+		return oldPVC, fmt.Errorf("can't patch status of  PVC %s with %v", klog.KObj(oldPVC), updateErr)
+	}
+	err = ctrl.claims.Update(updatedClaim)
+	if err != nil {
+		return oldPVC, fmt.Errorf("error updating PVC %s in local cache: %v", klog.KObj(newPVC), err)
+	}
+
+	return updatedClaim, nil
+}
+
+func (ctrl *modifyController) patchPersistentVolume(oldPV, newPV *v1.PersistentVolume) (*v1.PersistentVolume, error) {
+	patchBytes, err := util.GetPatchData(oldPV, newPV)
+	if err != nil {
+		return nil, fmt.Errorf("can't update capacity of PV %s as generate path data failed: %v", newPV.Name, err)
+	}
+	updatedPV, updateErr := ctrl.kubeClient.CoreV1().PersistentVolumes().Patch(context.TODO(), newPV.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
+	if updateErr != nil {
+		return nil, fmt.Errorf("update capacity of PV %s failed: %v", newPV.Name, updateErr)
+	}
+	err = ctrl.volumes.Update(updatedPV)
```

Comment:
and here.

Reply:
Oh I think we only need to patch the PV but not the PVC. I was following the resizer implementation, because the capacity update is updated there so we should keep it the same for modify volume operations.